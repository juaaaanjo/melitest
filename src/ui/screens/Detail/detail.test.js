import React from 'react';
import { render } from '@testing-library/react-native';
import DetailScreen from './Detail';

jest.mock('@react-navigation/native');

describe('DetailScreen', () => {
  const mockRouteParams = {
    idProduct: '123',
    title: 'Product Title',
    price: 100,
    thumbnail: 'https://example.com/image.jpg',
    seller: 'Seller Name',
    attributes: [],
    condition: 'new',
    available_quantity: 10,
  };

  test('renders correctly', () => {
    const { toJSON } = render(<DetailScreen route={{ params: mockRouteParams }} />);
    expect(toJSON()).toMatchSnapshot();
  });

  test('displays correct product information', () => {
    const { getByText } = render(<DetailScreen route={{ params: mockRouteParams }} />);
    expect(getByText(mockRouteParams.title)).toBeTruthy();
    expect(getByText(`$${mockRouteParams.price.toLocaleString()}`)).toBeTruthy();
    expect(getByText(mockRouteParams.seller)).toBeTruthy();
    expect(getByText('Stock disponible')).toBeTruthy();
    expect(getByText('Condici√≥n')).toBeTruthy();
  });

});