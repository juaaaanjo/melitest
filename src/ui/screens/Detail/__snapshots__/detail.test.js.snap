// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailScreen renders correctly 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "backgroundColor": "white",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
      },
      {
        "marginTop": -66.7,
      },
    ]
  }
>
  <View
    style={
      {
        "flex": 1,
        "paddingHorizontal": 45,
      }
    }
  >
    <RCTScrollView
      showsVerticalScrollIndicator={false}
    >
      <View>
        <View
          style={
            {
              "marginBottom": undefined,
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": 66.7,
            }
          }
          testID="child"
        >
          <Image
            source={
              {
                "uri": "https://example.com/image.jpg",
              }
            }
            style={
              {
                "alignSelf": "center",
                "height": 250,
                "width": 250,
              }
            }
          />
        </View>
        <View
          style={
            {
              "marginBottom": undefined,
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": 40.019999999999996,
            }
          }
          testID="child"
        >
          <Text
            collapsable={false}
            style={
              {
                "alignSelf": "center",
                "backgroundColor": "transparent",
                "color": "#000000",
                "fontSize": 20,
                "fontStyle": "normal",
                "fontWeight": "600",
                "marginTop": 0,
                "paddingLeft": 0,
                "textAlign": "left",
                "transform": [
                  {
                    "translateY": 0,
                  },
                ],
                "width": 350,
              }
            }
          >
            Product Title
          </Text>
        </View>
        <View
          style={
            {
              "marginBottom": undefined,
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": 13.34,
            }
          }
          testID="child"
        >
          <Text
            collapsable={false}
            style={
              {
                "alignSelf": "center",
                "backgroundColor": "transparent",
                "color": "#555",
                "fontSize": 20,
                "fontStyle": "normal",
                "fontWeight": "300",
                "marginTop": 0,
                "paddingLeft": 0,
                "textAlign": "left",
                "transform": [
                  {
                    "translateY": 0,
                  },
                ],
                "width": 350,
              }
            }
          >
            $
            100
          </Text>
        </View>
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 26.68,
            }
          }
        >
          <View>
            <Text
              collapsable={false}
              style={
                {
                  "alignSelf": "center",
                  "backgroundColor": "transparent",
                  "color": "#000000",
                  "fontSize": 12,
                  "fontStyle": "normal",
                  "fontWeight": "700",
                  "marginTop": 0,
                  "paddingLeft": 0,
                  "textAlign": "left",
                  "transform": [
                    {
                      "translateY": 0,
                    },
                  ],
                  "width": 150,
                }
              }
            >
              Stock disponible
            </Text>
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginLeft": 37.5,
                  "marginTop": 26.68,
                }
              }
            >
              <Text />
              <View
                style={
                  {
                    "marginBottom": undefined,
                    "marginLeft": 15,
                    "marginRight": undefined,
                    "marginTop": undefined,
                  }
                }
                testID="child"
              >
                <Text
                  collapsable={false}
                  style={
                    {
                      "alignSelf": "center",
                      "backgroundColor": "transparent",
                      "color": "#757575",
                      "fontSize": 12,
                      "fontStyle": "normal",
                      "fontWeight": "700",
                      "marginTop": 0,
                      "paddingLeft": 0,
                      "textAlign": "left",
                      "transform": [
                        {
                          "translateY": 0,
                        },
                      ],
                      "width": 150,
                    }
                  }
                >
                  10
                </Text>
              </View>
            </View>
          </View>
          <View>
            <Text
              collapsable={false}
              style={
                {
                  "alignSelf": "center",
                  "backgroundColor": "transparent",
                  "color": "#000000",
                  "fontSize": 12,
                  "fontStyle": "normal",
                  "fontWeight": "700",
                  "marginTop": 0,
                  "paddingLeft": 0,
                  "textAlign": "left",
                  "transform": [
                    {
                      "translateY": 0,
                    },
                  ],
                  "width": 150,
                }
              }
            >
              Condici√≥n
            </Text>
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginLeft": 37.5,
                  "marginTop": 26.68,
                }
              }
            >
              <Text />
              <View
                style={
                  {
                    "marginBottom": undefined,
                    "marginLeft": 15,
                    "marginRight": undefined,
                    "marginTop": undefined,
                  }
                }
                testID="child"
              >
                <Text
                  collapsable={false}
                  style={
                    {
                      "alignSelf": "center",
                      "backgroundColor": "transparent",
                      "color": "#757575",
                      "fontSize": 12,
                      "fontStyle": "normal",
                      "fontWeight": "700",
                      "marginTop": 0,
                      "paddingLeft": 0,
                      "textAlign": "left",
                      "transform": [
                        {
                          "translateY": 0,
                        },
                      ],
                      "width": 150,
                    }
                  }
                >
                  Nuevo
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "marginBottom": undefined,
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": 26.68,
            }
          }
          testID="child"
        >
          <Text
            collapsable={false}
            style={
              {
                "alignSelf": "center",
                "backgroundColor": "transparent",
                "color": "#000000",
                "fontSize": 18,
                "fontStyle": "normal",
                "fontWeight": "700",
                "marginTop": 0,
                "paddingLeft": 0,
                "textAlign": "left",
                "transform": [
                  {
                    "translateY": 0,
                  },
                ],
                "width": 350,
              }
            }
          >
            Vendido por
          </Text>
        </View>
        <View
          style={
            {
              "marginBottom": undefined,
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": 13.34,
            }
          }
          testID="child"
        >
          <Text
            collapsable={false}
            style={
              {
                "alignSelf": "center",
                "backgroundColor": "transparent",
                "color": "#00B2FF",
                "fontSize": 20,
                "fontStyle": "normal",
                "fontWeight": "300",
                "marginTop": 0,
                "paddingLeft": 0,
                "textAlign": "left",
                "transform": [
                  {
                    "translateY": 0,
                  },
                ],
                "width": 350,
              }
            }
          >
            Seller Name
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
